src/construction.cpp

```diff
@@ -738,14 +733,23 @@ bool player_can_build( player &p, const inventory &inv, const std::string &desc
 
 bool player_can_build( player &p, const inventory &inv, const construction &con )
 {
+
     if( p.has_trait( trait_DEBUG_HS ) ) {
         return true;
     }
 
     if( !p.meets_skill_requirements( con ) ) {
         return false;
     }
-    return con.requirements->can_make_with_inventory( inv, is_crafting_component );
+
+    const bool can_build = con.requirements->can_make_with_inventory( inv, is_crafting_component );
+    if( !can_build ) {
+        popup( _( "You can't build that!" ) );
```

Comment:
This is probably the culprit of #38266. `player_can_build` is used for generating the list of available constructions and shouldn't call any UI or input code.